src/runtime/bindle.rs

```diff
@@ -222,6 +271,94 @@ fn parcel_url(bindle_id: &Id, parcel_sha: String) -> String {
     )
 }
 
+/// Given a bindle's invoice and parcel directory, build a module configuration.
+pub async fn standalone_invoice_to_modules(
+    invoice: &Invoice,
+    parcel_dir: PathBuf,
+    asset_cache: PathBuf,
+) -> anyhow::Result<ModuleConfig> {
+    let mut modules = IndexSet::new();
+    let bindle_id = invoice.bindle.id.clone();
+
+    // For each top-level entry, if it is a Wasm module, we create a Module.
+    let top = top_modules(invoice);
+    debug!(
+        default_modules = top.len(),
+        "Loaded modules from the default group (parcels that do not have conditions.memberOf set)"
+    );
+
+    for parcel in top {
+        // Create a basic module definition from the features section on this parcel.
+        let mut def = wagi_features(&invoice.bindle.id, &parcel);
+
+        def.module = parcel_dir
+            .join(format!("{}.dat", parcel.label.sha256))
+            .to_string_lossy()
+            .to_string();
+
+        // If the parcel has a group, get the group.
+        // Then we have to figure out how to map the group onto a Wagi configuration.
+        if let Some(c) = parcel.conditions.clone() {
+            let groups = c.requires.unwrap_or_default();
+            for n in groups.iter() {
```

Comment:
Nit: I think you own all the data here, so into_iter will save you the clone below